src/components/Amplitude.js

```diff
@@ -1,149 +1,43 @@
-    }
-  }
-
-  componentDidMount() {
-    const { props } = this;
-    const amplitudeInstance = this.getAmplitudeInstance();
-
-    if (amplitudeInstance && props.userProperties) {
-      amplitudeInstance.setUserProperties(props.userProperties);
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    this.instrument.cache.garbageCollect();
-
-    const { props } = this;
-    const amplitudeInstance = this.getAmplitudeInstance();
-
-    if (amplitudeInstance && !shallowequal(prevProps.userProperties, props.userProperties)) {
-      amplitudeInstance.setUserProperties(props.userProperties);
-    }
-  }
+import PropTypes from 'prop-types';
 
-  getChildContext() {
+import { AmplitudeContext } from './AmplitudeProvider';
+import { useInstrument } from '../hooks/use-instrument';
+import { useLogEvent } from '../hooks/use-log-event';
+import { useSetUserProperties } from '../hooks/use-set-user-properties';
+
+const Amplitude = ({
+  children,
+  debounceInterval,
+  eventProperties,
+  instanceName = '$default_instance',
+  userProperties,
+}) => {
+  const logEvent = useLogEvent(instanceName, eventProperties, debounceInterval);
+  const instrument = useInstrument(instanceName, eventProperties, debounceInterval);
+
+  useSetUserProperties(instanceName, userProperties);
+
+  const inheritedContext = React.useContext(AmplitudeContext);
+  const contextValue = React.useMemo(() => {
     return {
-      getAmplitudeEventProperties: this.getAmplitudeEventProperties,
+      ...inheritedContext,
+      eventProperties: { ...inheritedContext.eventProperties, ...eventProperties },
     };
-  }
-
-  getAmplitudeInstance = () => {
-    const { context, props } = this;
-
-    if (!context.getAmplitudeInstance) {
-      return null;
-    }
-
-    const amplitudeInstance = context.getAmplitudeInstance(props.instanceName);
-
-    if (!isValidAmplitudeInstance(amplitudeInstance)) {
-      console.error(
-        'Failed to get a valid Amplitude instance. This likely means the "amplitudeInstance" prop your provided to the AmplitudeProvider component is not a valid Amplitude instance.',
-      );
-
-      return null;
-    }
-
-    return amplitudeInstance;
-  };
+  }, [eventProperties, inheritedContext]);
 
-  getAmplitudeEventProperties = () => {
-    const { props, context } = this;
+  let childElements;
 
-    if (!context.getAmplitudeEventProperties) {
-      return props.eventProperties;
-    }
-
-    const inheritedEventProperties = context.getAmplitudeEventProperties();
-
-    if (typeof props.eventProperties === 'function') {
-      return props.eventProperties(inheritedEventProperties);
-    } else {
-      return {
-        ...inheritedEventProperties,
-        ...props.eventProperties,
-      };
-    }
-  };
-
-  render() {
-    const { props } = this;
-
-    if (typeof props.children === 'function') {
-      return props.children(this._renderPropParams);
-    } else {
-      return props.children || null;
-    }
+  if (typeof children === 'function') {
```

Comment:
i guess this was working the same way before - but it's confusing for me that this component actually has 2 roles:
- overriding/shadowing AmplitudeContext value
- acting as render prop amplitude provider

IMHO it should only focus on the latter, as the former responsibility is already covered (or should be) by `AmplitudeProvider` component.